### PR TITLE
Fix dtype explanation in reduce ops docs

### PR DESCRIPTION
**Pull request by zou3519**
Followup to #7654

cc @gchanan

### CODE REVIEW

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -21,9 +21,11 @@ def parse_kwargs(desc):
 
 
 reduceops_common_args = parse_kwargs("""
-    dtype (:class:`torch.dtype`, optional): the desired data type of returned tensor.
-        If specified, the input tensor is casted to :attr:`dtype` before the operation
-        is performed. This is useful for preventing data type overflows. Default: None.
+    dtype (:class:`torch.dtype`, optional): the desired data type of the returned
+        tensor and of the accumulator in which the elements are reduced. If ``None``,
+        then uses ``input.dtype``.
```
Comment (gchanan) [inline torch/_torch_docs.py]: Sorry, should have mentioned this in the previous PR: the 'None' explanation is also incorrect, see the "Integer Upcast" section of https://github.com/gchanan/pytorch/wiki/Reduction-dtype-notes.

**Comment by ezyang** [comment]
@zou3519 Don't forget about this PR!

**Comment by yf225** [comment]
Ping @zou3519

**Comment by zou3519** [comment]
Will push an update to this soon. Action items are:
1) add an upcasting section to docs, if it doesn't exist already
2) link to it here for the case where dtype is None.

**Comment by zou3519** [comment]
I don't remember the context behind this so I am closing it

**Comment by gchanan** [comment]
I think the context was to describe the semantics, not the implementation details.

**Pull request closed without merging**